### PR TITLE
fix(orchestrator):workflow name column values in workflow runs table …

### PR DESCRIPTION
**Pull request by LiorSoffer**
resolve [FLPATH-1967](https://issues.redhat.com/browse/FLPATH-1967)
image

### CODE REVIEW

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowDetailsTabContent.tsx [hunk 1]
```diff
@@ -55,7 +51,7 @@ export const WorkflowDetailsTabContent = ({
             <WorkflowEditor
               kind={EditorViewKind.EXTENDED_DIAGRAM_VIEWER}
               workflowId={workflowId}
-              format={workflowFormat}
+              format={workflowOverviewDTO?.format}
```
Comment (gciavarrini) [inline workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowDetailsTabContent.tsx:54]: what if `workflowOverviewDTO` is `undefined`?

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx [hunk 2]
```diff
@@ -243,7 +243,6 @@ const RefForwardingWorkflowEditor: ForwardRefRenderFunction<
             if (format && workflowId && format !== workflowFormat) {
               const link = viewWorkflowLink({
                 workflowId: workflowId,
-                format: workflowFormat,
```
Comment (gciavarrini) [inline workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx:246]: do we stil need `format` in the `if` condition?

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx [hunk 3]
```diff
@@ -243,7 +243,6 @@ const RefForwardingWorkflowEditor: ForwardRefRenderFunction<
             if (format && workflowId && format !== workflowFormat) {
               const link = viewWorkflowLink({
                 workflowId: workflowId,
```
Comment (batzionb) [inline workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx:245]: I'm pretty sure we can't just remove the format
Checking now why it's needed

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowDetailsTabContent.tsx [hunk 4]
```diff
@@ -35,11 +35,7 @@ export const WorkflowDetailsTabContent = ({
   loading,
   workflowOverviewDTO,
 }: Props) => {
-  const { workflowId, format } = useRouteRefParams(workflowRouteRef);
-  const workflowFormat = useMemo(
-    () => (format === 'json' ? 'json' : 'yaml'),
-    [format],
-  );
+  const { workflowId } = useRouteRefParams(workflowRouteRef);
```
Comment (batzionb) [inline workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowDetailsTabContent.tsx:38]: small improvement suggestion:
we can also take the workflow id from workflowOverviewDTO
this way the component becomes completely independent from the routing system, making it more reusable and isolated
WDYT?

**Comment by batzionb** [comment]
@LiorSoffer 
I like this solution
It makes the links more concise and easy to use
Thanks!

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx (hunk 3 above)
Comment (batzionb) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx:245]: I see you take it from WorkflowOverviewDTO which is the better approach :)

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx (hunk 2 above)
Comment (batzionb) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx:246]: good point, please remove the format from the condition

Comment (batzionb) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx:246]: there's some kind of odd rerouting here...
this code handles a case where workflowFormat is different than the format in the URL
if this is the case, rerouting won't help
this whole code should remove

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowDetailsTabContent.tsx (hunk 1 above)
Comment (batzionb) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowDetailsTabContent.tsx:54]: I think the whole page should not show anything if workflowOverviewFormatDTO is undefined
It should be handled in WorkflowPage

Comment (batzionb) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowDetailsTabContent.tsx:54]: @LiorSoffer 
WDYT of WorkflowOverviewDTO property being mandatory
In case it's undefined WorkflowPage should show an error and no content

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx (hunk 2 above)
Comment (batzionb) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx:246]: if the formats don't match we need to handle it differently
maybe show an error

Comment (batzionb) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowEditor/WorkflowEditor.tsx:246]: Took a look where format comes from
Backend calculates it
https://github.com/redhat-developer/rhdh-plugins/blob/main/workspaces/orchestrator/plugins/orchestrator-backend/src/service/SonataFlowService.ts#L194
Now that it comes from the backend, recalculating it in the FE is a waste
This whole code should be removed, including recalculating the format and handling the mismatch

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowDetailsTabContent.tsx (hunk 4 above)
Comment (LiorSoffer) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowDetailsTabContent.tsx:38]: good idea

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowPage.tsx [hunk 5]
```diff
@@ -66,10 +66,12 @@ export const WorkflowPage = () => {
             loading={loading}
             canRun={canRun}
           >
-            <WorkflowDetailsTabContent
-              loading={loading}
-              workflowOverviewDTO={workflowOverviewDTO?.data}
-            />
+            {workflowOverviewDTO && (
```
Comment (batzionb) [inline workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowPage.tsx:69]: actually this might not work well, I think when in loading state workflowOverviewDTO is undefined, and this way we won't show the loading state (skeletons) that we have

Comment (batzionb) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowPage/WorkflowPage.tsx:69]: maybe better pass it, and add a more simple loading handling, if loading show <Progress>
if workflowOverviewDTO not defined, show null
make these changes in WorkflowDetailsTabContent and not here

**Review by batzionb — DISMISSED** [review]

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx [hunk 6]
```diff
@@ -189,35 +191,37 @@ export const WorkflowRunsTabContent = () => {
...
-                capitalize(row.category ?? VALUE_UNAVAILABLE),
-              sorting: false,
+              render: (data: WorkflowRunDetail) =>
+                capitalize(data.category ?? VALUE_UNAVAILABLE),
             },
           ]),
-      { title: 'Started', field: 'start', customSort: applyBackendSort },
-      { title: 'Duration', field: 'duration', sorting: false },
+      { title: 'Started', field: 'start', defaultSort: 'desc' },
+      { title: 'Duration', field: 'duration' },
```
Comment (batzionb) [inline workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx:222]: please return sorting: false
sorting is not possible for duration

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx [hunk 7]
```diff
@@ -189,35 +191,37 @@ export const WorkflowRunsTabContent = () => {
...
-              render: (row: WorkflowRunDetail) =>
-                capitalize(row.category ?? VALUE_UNAVAILABLE),
-              sorting: false,
+              render: (data: WorkflowRunDetail) =>
+                capitalize(data.category ?? VALUE_UNAVAILABLE),
             },
           ]),
-      { title: 'Started', field: 'start', customSort: applyBackendSort },
-      { title: 'Duration', field: 'duration', sorting: false },
+      { title: 'Started', field: 'start', defaultSort: 'desc' },
```
Comment (batzionb) [inline workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx:221]: undo this change as well

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx (hunk 6 above)
Comment (LiorSoffer) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx:222]: should I return customSort: applyBackendSort  as well?

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx (hunk 7 above)
Comment (batzionb) [reply workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx:221]: it should be customSort: applyBackendSort

File: workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx [hunk 8]
```diff
@@ -191,33 +193,36 @@ export const WorkflowRunsTabContent = () => {
...
         ? []
         : [
             {
               title: 'Category',
               field: 'category',
-              render: (row: WorkflowRunDetail) =>
-                capitalize(row.category ?? VALUE_UNAVAILABLE),
-              sorting: false,
+              render: (data: WorkflowRunDetail) =>
+                capitalize(data.category ?? VALUE_UNAVAILABLE),
```
Comment (batzionb) [inline workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx:219]: return the sorting: false

**Review by batzionb — APPROVED** [review]
